File: drills/ts/harnesses/nc51_counting_bits.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.countBits !== "function") throw new Error("__signature_mismatch__");
  return [
    ["countBits(5)", show([0, 1, 1, 2, 1, 2]), show(solution.countBits(5))],
    ["countBits(2)", show([0, 1, 1]), show(solution.countBits(2))],
    ["countBits(0)", show([0]), show(solution.countBits(0))],
    ["countBits(8)", show([0, 1, 1, 2, 1, 2, 2, 3, 1]), show(solution.countBits(8))],
  ];
}

File: drills/go/harnesses/nc51_counting_bits.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("countBits(2)", []int{0, 1, 1}, countBits(2)),
			tc("countBits(5)", []int{0, 1, 1, 2, 1, 2}, countBits(5)),
			tc("countBits(0)", []int{0}, countBits(0)),
			tc("countBits(8)", []int{0, 1, 1, 2, 1, 2, 2, 3, 1}, countBits(8)),
		}
	})
}
